### DISCUSSION TITLE
family option does not work as expected.

### DISCUSSION THREAD

**Discussion by RHEF19** [ImageMagick/ImageMagick#2308] [Help]
As an example, the following command line will produce exactly the same output.
```
magick -font "MS-Gothic-&-MS-UI-Gothic-&-MS-PGothic"                      -pointsize 32 label:MMMM\nIIII test1.png
magick -font "MS-Gothic-&-MS-UI-Gothic-&-MS-PGothic" -family "MS Gothic"  -pointsize 32 label:MMMM\nIIII test2.png
magick -font "MS-Gothic-&-MS-UI-Gothic-&-MS-PGothic" -family "MS PGothic" -pointsize 32 label:MMMM\nIIII test3.png
```
"MS Gothic" is a monospaced font and "MS PGothic" is a proportional font.
And I want the result of processing by this proportional font, but I can't get it.
There are no errors or anything, but label, annotate and draw have the same result.
Is there something wrong with this command line?

list font
```
...
  Font: MS-Gothic-&-MS-UI-Gothic-&-MS-PGothic
    family: MS Gothic & MS UI Gothic & MS PGothic
    style: Normal
    stretch: Normal
    weight: 400
    glyphs: c:\windows\fonts\msgothic.ttc
...
```
version
```
Version: ImageMagick 7.0.10-24 Q16 x64 2020-07-18 http://www.imagemagick.org
Copyright: Copyright (C) 1999-2018 ImageMagick Studio LLC
License: http://www.imagemagick.org/script/license.php
Visual C++: 192628806
Features: Cipher DPC OpenCL
Delegates (built-in): bzlib cairo flif freetype gslib heic jng jp2 jpeg lcms lqr lzma openexr pangocairo png ps raw rsvg tiff webp xml zlib
```

**Comment by snibgo** [comment]
If the family is "MS Gothic & MS UI Gothic & MS PGothic", that is what you need with "-family".

**Reply by RHEF19** [reply]
The result was no different.
What I want is a result of "MS PGothic".

**Comment by snibgo** [comment]
According to your "-list font", you have no font or family called "MS PGothic".

**Reply by RHEF19** [reply]
This is a Japanese font, so if you don't have Japanese installed on your system, it may not be listed.
The following fonts are also fine.

```
  Font: Cambria-&-Cambria-Math
    family: Cambria & Cambria Math
    style: Normal
    stretch: Normal
    weight: 400
    glyphs: c:\windows\fonts\cambria.ttc
```
```
magick -font "Cambria-&-Cambria-Math"                        -pointsize 32 label:MMMM\nIIII test1.png
magick -font "Cambria-&-Cambria-Math" -family "Cambria"      -pointsize 32 label:MMMM\nIIII test2.png
magick -font "Cambria-&-Cambria-Math" -family "Cambria Math" -pointsize 32 label:MMMM\nIIII test3.png
```
These are the same results.

**Comment by snibgo** [comment]
Those three commands give the same result because IM ignores your "-family".

If you ask for a font that you have, IM will use that font. If you also specify a family, that will be used _only_ if the font you asked for isn't available.

**Reply by RHEF19** [reply]
```
magick -family "Cambria"      -pointsize 32 label:MMMM\nIIII test2.png
magick -family "Cambria Math" -pointsize 32 label:MMMM\nIIII test3.png
```
This example can be run without error, but the result is the same.

Please show me an example of using "Cambria Math" in the font "Cambria-&-Cambria-Math".

**Comment by snibgo** [comment]
I don't understand your question. If you specify a family with no specific font, IM will try to find a font in that family. If there aren't any, IM will use some other font (eg "Arial").

My computer has fonts in the "Cambria" family but none in the "Cambria Math" family. For example:
```
magick -list font |grep -i cambria

  Font: Cambria-&-Cambria-Math
    family: Cambria & Cambria Math
    glyphs: c:\windows\fonts\cambria.ttc
  Font: Cambria-Bold
    family: Cambria
    glyphs: c:\windows\fonts\cambriab.ttf
  Font: Cambria-Bold-Italic
    family: Cambria
    glyphs: c:\windows\fonts\cambriaz.ttf
  Font: Cambria-Italic
    family: Cambria
    glyphs: c:\windows\fonts\cambriai.ttf```
Your computer may have different fonts.

**Reply by RHEF19** [reply]
```
  Font: Cambria-&-Cambria-Math
    family: Cambria & Cambria Math
                      ^^^^^^^^^^^^
```
I'm talking about this one of yours.

For example, if you set the notepad font to "Cambria Math", you can use that font.
You can try it.
The result should be different from "Cambria".

I want to know how to use "Cambria Math" in ImageMagick.


By the way, my computer:
```
  Font: Cambria-&-Cambria-Math
    family: Cambria & Cambria Math
    style: Normal
    stretch: Normal
    weight: 400
    glyphs: c:\windows\fonts\cambria.ttc
  Font: Cambria-Bold
    family: Cambria
    style: Normal
    stretch: Normal
    weight: 700
    glyphs: c:\windows\fonts\cambriab.ttf
  Font: Cambria-Bold-Italic
    family: Cambria
    style: Italic
    stretch: Normal
    weight: 700
    glyphs: c:\windows\fonts\cambriaz.ttf
  Font: Cambria-Italic
    family: Cambria
    style: Italic
    stretch: Normal
    weight: 400
    glyphs: c:\windows\fonts\cambriai.ttf
```

**Comment by snibgo** [comment]
> For example, if you set the notepad font to "Cambria Math", you can use that font.
You can try it.
The result should be different from "Cambria".

On my computer, in Notebook, I can use "Cambria" or "Cambria Math" fonts. They seem to be identical.

Control Panel, Fonts, Properties says fonts "Cambria Regular" and "Cambria Math Regular" are both from the cambria.ttc file.

**Reply by RHEF19** [reply]
Even if the files are identical, they are usually treated as separate fonts.
The shape of the characters, character spacing and line spacing are also different.

Did you look "identical" when they were so different?
![example](https://user-images.githubusercontent.com/49441484/88352129-ad52e900-cd93-11ea-946a-8ff5dd5154c7.png)

**Comment by snibgo** [comment]
According to https://en.wikipedia.org/wiki/TrueType : "TrueType Collection (TTC) is an extension of TrueType format that allows combining multiple fonts into a single file, creating substantial space savings for a collection of fonts with many glyphs in common."

IM uses the freetype library to process fonts. Perhaps this library incorrectly assumes that each font file has only one font. Or perhaps that assumption is made in ImageMagick.

A developer may comment.

**Reply by RHEF19** [reply]
Yes.
I think it's best to have someone who is familiar with the inner workings of that function look at it, too.
I will continue to await answers.
Thank you.

**Comment by snibgo** [comment]
A possible workaround is to use a tool that can separate the fonts in a TTC into individual TTF files. For example, see https://stackoverflow.com/questions/15455895/convert-or-extract-ttc-font-to-ttf-how-to

**Reply by RHEF19** [reply]
Yes.
That's how I'm doing it now.
However, I was hoping that using "family" option would solve the problem, as there are some cases where file modification is not allowed.
